adafruit_esp32spi/adafruit_esp32spi_wifimanager.py

```diff
@@ -76,21 +95,45 @@ def connect(self):
             for access_pt in self._esp.scan_networks():
                 print("\t%s\t\tRSSI: %d" % (str(access_pt['ssid'], 'utf-8'), access_pt['rssi']))
         failure_count = 0
-        while not self._esp.is_connected:
-            try:
-                if self.debug:
-                    print("Connecting to AP...")
-                self.pixel_status((100, 0, 0))
-                self._esp.connect_AP(bytes(self.ssid, 'utf-8'), bytes(self.password, 'utf-8'))
-                failure_count = 0
-                self.pixel_status((0, 100, 0))
-            except (ValueError, RuntimeError) as error:
-                print("Failed to connect, retrying\n", error)
-                failure_count += 1
-                if failure_count >= self.attempts:
+        if self.wificonntype == WiFiConnType.normal:
+            while not self._esp.is_connected:
+                try:
+                    if self.debug:
+                        print("Connecting to AP...")
+                    self.pixel_status((100, 0, 0))
+                    self._esp.connect_AP(bytes(self.ssid, 'utf-8'), bytes(self.password, 'utf-8'))
+                    failure_count = 0
+                    self.pixel_status((0, 100, 0))
+                except (ValueError, RuntimeError) as error:
+                    print("Failed to connect, retrying\n", error)
+                    failure_count += 1
+                    if failure_count >= self.attempts:
+                        failure_count = 0
+                        self.reset()
+                    continue
+        elif self.wificonntype == WiFiConnType.enterprise:
+            self._esp.wifi_set_network(bytes(self.ent_ssid, 'utf-8'))
+            self._esp.wifi_set_entidentity(bytes(self.ent_ident, 'utf-8'))
+            self._esp.wifi_set_entusername(bytes(self.ent_user, 'utf-8'))
+            self._esp.wifi_set_entpassword(bytes(self.ent_passwd, 'utf-8'))
+            self._esp.wifi_set_entenable()
+            while not self._esp.is_connected:
+                try:
+                    if self.debug:
+                        print("Connecting to WPA2 Enterprise AP...")
+                    self.pixel_status((100, 0, 0))
                     failure_count = 0
-                    self.reset()
-                continue
+                    self.pixel_status((0, 100, 0))
+                except (ValueError, RuntimeError) as error:
+                    print("Failed to connect, retrying\n", error)
+                    failure_count += 1
+                    if failure_count >= self.attempts:
+                        failure_count = 0
+                        self.reset()
+                    continue
+        else:
+            print("Invalid connection type! Exiting...")
```

Comment:
CircuitPython libraries don't normally `print` and `exit`. They usually `raise` an error. Instead of L135/L136,

`raise TypeError('Invalid wificonntype specified.')`

Reply:
I was thinking there has to be a better way/more Python way of dealing with this. And also realized I never tested this code. Fixed in an upcoming commit.